Add more robust method to retrieve config JSON from the terminal

stereokai opened the pull request:
@timolins 
This fixes [Hyperlayout simply printing out config](#8) for me and gets Hyperlayout to work on Windows. 😄

timolins commented on index.js:
```diff
@@ -154,12 +154,19 @@ exports.middleware = store => next => action => {
   const {sessions} = store.getState()
   const {activeUid} = sessions
+        // Remove ANSI escape code sequences. Visualization: https://goo.gl/IY8vuU
+  const ANSI_escape_codes = /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/gm
```
These constants won't change, so they shouldn't be defined inside the middleware.

stereokai replied:
That's true.

timolins commented:
Awesome! Would be cool if you changed the variables from snake_case to camelCase to obey the linter. 🐫

timolins commented:
Thanks 🙏

stereokai commented:
Green light

timolins commented:
This might sound weird, but your pull request basically killed my Hyper.app. 😅
![crash](https://cloud.githubusercontent.com/assets/1440854/21201644/0f4bba8a-c24c-11e6-810f-8f2886344e00.png)

stereokai commented:
:(

Maybe try to disable one or more of the regexes I introduced to discover the culprit?

stereokai commented:
@timolins I had the same problem with PowerShell in my machine. (I tried with 3 different shells, including cmd and native Bash). The problem with PowerShell was that it was printing a security error to the terminal **immediately after it was loaded** - and it threw hyperlayout into a loop or a freeze somehow. Just to be more clear - that error text was being printed to the terminal before PowerShell's command prompt was visible and ready to receive commands.

That error was not related to Hyper, and would happen even if I opened PowerShell independently. The moment I fixed the error, (read: removed the content that was being printed before the command prompt was available in the terminal) the freeze was gone.

Does your terminal print some things before the command prompt is ready for input?

stereokai commented:
@timolins still there? :)

timolins commented:
Awesome, thank you! Don't know how I missed this... Will check this out soon!

timolins commented:
And no, my command prompt doesn't print anything before it's ready for input.

stereokai commented:
@timolins Did you try to disable one or more of the regexes I introduced to discover the culprit like I suggested? :)

colinbate commented:
This code killed my Hyper too. It is the `configMatcher` which does it.

I've been wrestling with getting hyperlayout to work at all on Windows 7. The addition of the ANSI codes and line breaks isn't a simple one to solve. For example, my config was broken onto three lines, the first line break being inserted **instead** of a space within a command string. The second was inserted mid-string. If I replace the newlines with an empty string, I break that first command.

colinbate commented:
If I change the size of my Hyper window, then the line breaks go elsewhere and can be removed more safely. Is this really the best way to communicate between the command and the plugin?

timolins commented:
I'm open for suggestions. While developing hyperlayout, I tried a few different approaches, but the current implementation was the best one I came up with. Of course this doesn't mean that there isn't a better one. Feel free to come up with a new solution! 🙂

colinbate commented:
Fair enough. At the moment I feel frustrated by the lack of documentation around the Hyper APIs. :) I don't think the issue is with hyperlayout per se. 

Would it be possible for the plugin to do some of the work the command does? Reading the config files?

iamstarkov commented:
this fix seems to solve problems on windows

philoserf commented:
I see the project has drifted from the pull request.
